File: CodeChef/SnackDown/PreB/SNCOUP.cpp
#include <iostream>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    char a[2][100005];
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        cin >> a[0];
        cin >> a[1];

        int count = 1;
        int pendingTop = 0;
        int pendingBottom = 0;
        int upCount = 0;
        int DownCount = 0;

        for (int j = 0; j < n ; j++) {
            
            if (a[0][j] == '.' && a[1][j] == '.')
                continue;

            if (a[0][j] == '*' && a[1][j] == '.') {
                upCount++;
                if (!pendingTop) {
                    pendingTop = 1;
                    continue;
                }
                pendingBottom = 0;
                count++;
                continue;
            }

            if (a[0][j] == '.' && a[1][j] == '*') {
                DownCount++;
                if (!pendingBottom) {
                    pendingBottom = 1;
                    continue;
                }
                pendingTop = 0;
                count++;
                continue;
            }

            if (a[0][j] == '*' && a[1][j] == '*') {
                upCount++;
                DownCount++;
                if (pendingBottom || pendingTop) {
                    count++;
                }
                pendingTop = 1;
                pendingBottom = 1;
                continue;
            }
        }
        if (upCount == 0 && DownCount == 0) {
            count = 0;
        }
        else if (upCount + DownCount == 1) {
            count = 0;
        }
        else if (upCount == 0 || DownCount == 0)
            count--;

        cout << count << endl;

    }

    return 0;
}

File: CodeChef/SnackDown/PreB/SNCOUP.py
t = int(raw_input().strip())
for i in xrange(t):
    n = int(raw_input().strip())
    a = []
    a.append(raw_input().strip())
    a.append(raw_input().strip())
    count = 1
    pendingTop = False
    pendingBottom = False
    upCount = 0
    DownCount = 0
    bruteChoice = 0
    for j in xrange(n):
        if a[0][j] == '.' and a[1][j] == '.':
                continue

        if a[0][j] == '*' and a[1][j] == '.': 
            upCount += 1
            if pendingTop or pendingBottom:
                bruteChoice += 1
            if not pendingTop: 
                pendingTop = True
                continue
            
            pendingBottom = False
            count += 1
            continue
        

        if a[0][j] == '.' and a[1][j] == '*':
            DownCount += 1
            if pendingTop or pendingBottom:
                bruteChoice += 1
            if not pendingBottom:
                pendingBottom = True
                continue
            
            pendingTop = False
            count += 1
            continue
        

        if a[0][j] == '*' and a[1][j] == '*': 
            upCount += 1
            DownCount += 1
            bruteChoice += 1
            if pendingBottom or pendingTop: 
                count += 1
                continue
            
            pendingTop = True
            pendingBottom = True
            continue
        
        
    if upCount == 0 and DownCount == 0: 
        count = 0
    
    elif upCount + DownCount == 1: 
        count = 0
    
    elif upCount == 0 or DownCount == 0:
        count -= 1

    print min(count, bruteChoice)